examples/cosmos/chain_miscellaneous_test.go

```diff
@@ -98,3 +127,204 @@ func BuildDependencies(ctx context.Context, t *testing.T, chain *cosmos.CosmosCh
+	require.NoError(t, err)
+
+	sendAmt := int64(1)
+	_, err = sendTokens(ctx, chain, users[0], users[1], "", sendAmt)
+	require.NoError(t, err)
+
+	b2New, err := chain.GetBalance(ctx, users[1].FormattedAddress(), chain.Config().Denom)
+	require.NoError(t, err)
+
+	require.Equal(t, b2.Add(math.NewInt(sendAmt)), b2New)
+}
+
+func testFindTxs(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, users []ibc.Wallet) {
+	height, _ := chain.Height(ctx)
+
+	_, err := sendTokens(ctx, chain, users[0], users[1], "", 1)
+	require.NoError(t, err)
+
+	txs, err := chain.FindTxs(ctx, height+1)
+	require.NoError(t, err)
+	require.NotEmpty(t, txs)
+	require.Equal(t, txs[0].Events[0].Type, "coin_spent")
+}
+
+func testPollForBalance(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, users []ibc.Wallet) {
+	bal2, err := chain.GetBalance(ctx, users[1].FormattedAddress(), chain.Config().Denom)
+	require.NoError(t, err)
+
+	amt := ibc.WalletAmount{
+		Address: users[1].FormattedAddress(),
+		Denom:   chain.Config().Denom,
+		Amount:  math.NewInt(1),
+	}
+
+	delta := uint64(3)
+
+	ch := make(chan error)
+	go func() {
+		new := amt
+		new.Amount = bal2.Add(math.NewInt(1))
+		ch <- cosmos.PollForBalance(ctx, chain, delta, new)
+	}()
+
+	err = chain.SendFunds(ctx, users[0].KeyName(), amt)
+	require.NoError(t, err)
+	require.NoError(t, <-ch)
+}
+
+func testRangeBlockMessages(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, users []ibc.Wallet) {
+	height, _ := chain.Height(ctx)
+
+	_, err := sendTokens(ctx, chain, users[0], users[1], "", 1)
+	require.NoError(t, err)
+
+	var bankMsgs []*banktypes.MsgSend
+	err = cosmos.RangeBlockMessages(ctx, chain.Config().EncodingConfig.InterfaceRegistry, chain.Validators[0].Client, height+1, func(msg sdk.Msg) bool {
+		found, ok := msg.(*banktypes.MsgSend)
+		if ok {
+			bankMsgs = append(bankMsgs, found)
+		}
+		return false
+	})
+	require.NoError(t, err)
+}
+
+func testAddingNode(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain) {
+	// This should be tested last or else Txs will fail on the new full node.
+	nodesAmt := len(chain.Nodes())
+	chain.AddFullNodes(ctx, nil, 1)
+	require.Equal(t, nodesAmt+1, len(chain.Nodes()))
+}
+
+func testBroadcaster(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, users []ibc.Wallet) {
+	from := users[0].FormattedAddress()
+	addr1 := "juno190g5j8aszqhvtg7cprmev8xcxs6csra7xnk3n3"
+	addr2 := "juno1a53udazy8ayufvy0s434pfwjcedzqv34q7p7vj"
+
+	c1 := sdk.NewCoins(sdk.NewCoin(chain.Config().Denom, math.NewInt(1)))
+	c2 := sdk.NewCoins(sdk.NewCoin(chain.Config().Denom, math.NewInt(2)))
+
+	b := cosmos.NewBroadcaster(t, chain)
+
+	in := banktypes.Input{
+		Address: from,
+		Coins:   c1.Add(c2[0]),
+	}
+	out := []banktypes.Output{
+		{
+			Address: addr1,
+			Coins:   c1,
+		},
+		{
+			Address: addr2,
+			Coins:   c2,
+		},
+	}
+
+	txResp, err := cosmos.BroadcastTx(
+		ctx,
+		b,
+		users[0],
+		banktypes.NewMsgMultiSend(in, out),
```

Comment:
Just for my own knowledge, could you explain a use case for this NewMultiMsgSend?

Reply:
This is useful in the future to use in the faucet.

I just wanted to ensure we tested non standard methods here for the broadcaster, and we do not yet use a MsgMultiSend anywhere. Just a good example for others on how to do it